app/database/server/models/channel.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+import {MM_TABLES} from '@constants/database';
+import field from '@nozbe/watermelondb/decorators/field';
+import children from '@nozbe/watermelondb/decorators/children';
+import Draft from '@typings/database/draft';
+import GroupsInChannel from '@typings/database/groups_in_channel';
+import PostInChannel from '@typings/database/post_in_channel';
+import MyChannelSettings from '@typings/database/my_channel_settings';
+import MyChannel from '@typings/database/my_channel';
+import ChannelInfo from '@typings/database/channel_info';
+import immutableRelation from '@nozbe/watermelondb/decorators/immutableRelation';
+import relation from '@nozbe/watermelondb/decorators/relation';
+import User from '@typings/database/user';
+import Team from '@typings/database/team';
+import ChannelMembership from '@typings/database/channel_membership';
+import Post from '@typings/database/post';
+
+export default class Channel extends Model {
+    static table = MM_TABLES.SERVER.CHANNEL
+    static associations: Associations = {
+        [MM_TABLES.SERVER.CHANNEL_INFO]: {type: 'has_many', foreignKey: 'channel_id'},
+        [MM_TABLES.SERVER.CHANNEL_MEMBERSHIP]: {type: 'has_many', foreignKey: 'channel_id'},
+        [MM_TABLES.SERVER.DRAFT]: {type: 'has_many', foreignKey: 'channel_id'},
+        [MM_TABLES.SERVER.GROUPS_IN_CHANNEL]: {type: 'has_many', foreignKey: 'channel_id'},
+        [MM_TABLES.SERVER.MY_CHANNEL]: {type: 'has_many', foreignKey: 'channel_id'},
+        [MM_TABLES.SERVER.MY_CHANNEL_SETTINGS]: {type: 'has_many', foreignKey: 'channel_id'},
```

Comment:
in this case a channel can only have one set of notifyProps as is only applicable to the current user, so this would be

```js
@relation(MM_TABLES.SERVER.MY_CHANNEL_SETTINGS, 'channel_id') settings;
```